src/tests/test_ncbi_taxonomy.py

```diff
@@ -0,0 +1,556 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of :mod:`ncbi_taxonomy` module.
+
+Typical usage example:
+
+    $ pytest test_ncbi_taxonomy.py
+
+"""
+
+from typing import ContextManager, Dict, List, Generator
+
+import pytest
+from pytest import param, raises
+from _pytest.fixtures import FixtureRequest
+from sqlalchemy.engine.url import make_url
+from sqlalchemy.exc import (
+    IntegrityError,
+    OperationalError,
+    SQLAlchemyError,
+    NoResultFound,
+)
+from sqlalchemy.ext.automap import automap_base
+from typing import ContextManager
+
+from ensembl.database import DBConnection, UnitTestDB
+from ensembl.ncbi_taxonomy.models import NCBITaxaName, NCBITaxaNode, NCBITaxonomy
+from ensembl.ncbi_taxonomy.api.utils import Taxonomy
+
+
+@pytest.mark.parametrize("db", [{"src": "ncbi_db"}], indirect=True)
+class TestNCBITaxonomyUtils:
+    """Tests :class: Taxonomy in utils.py"""
+
+    dbc = None  # type: UnitTestDB
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, db: UnitTestDB) -> None:
+        """Loads the required fixtures and values as class attributes.
+        Args:
+            request: Access to the requesting test context.
+            db: Generator of unit test database (fixture).
+        """
+        type(self).dbc = db.dbc
+
+    result_dict = {
+        "taxon_id": 9615,
+        "name": "beagle dog",
+        "name_class": "includes",
+        "parent_id": 9612,
+        "rank": "subspecies",
+        "genbank_hidden_flag": 1,
+        "left_index": 595,
+        "right_index": 596,
+        "root_id": 1,
+    }
+
+    @pytest.mark.parametrize("taxon_id, expectation", [(9615, result_dict)])
+    def test_fetch_node_by_id(self, taxon_id: int, expectation: NCBITaxonomy) -> None:
+        """Tests :func:`fetch_node_by_id()`.
+        Args:
+            taxon_id: An existing taxon_id as in ncbi_taxonomy (int)
+            expectation: :Class: NCBITaxonomy object.
+        """
+        with self.dbc.session_scope() as session:
+            taxonomy = Taxonomy()
```

Comment:
Since you implemented all methods as `class_method` no need to instantiate Taxonomy class at all. BTW: Make sure your class is `StateLess`

Reply:
```suggestion

```